src/main/java/org/jvnet/hudson/plugins/port_allocator/PortType.java

```diff
@@ -27,9 +27,7 @@
     public final String name;
 
     protected PortType(String name) {
-        // to avoid platform difference issue in case sensitivity of environment variables,
-        // always use uppser case.
-        this.name = name.toUpperCase();
+        this.name = name;
```

Comment:
> managed by a system property IMO

Property doesn't care about slave systems. Passing as is without doing magic is the best behaviour imho.